src-electron/util/types.js

```diff
@@ -78,6 +78,78 @@ async function typeSizeAttribute(db, zclPackageId, at, defaultValue = null) {
   }
 }
 
+/**
+ *
+ * @param {*} value
+ * @param {*} size
+ * @returns The negative hex value for a given positive hex value padded with
+ * the given size.
+ */
+function convertFloatToHex(value, size) {
+  let res = ''
+  const arrayBuffer = new ArrayBuffer(size)
+  const dataView = new DataView(arrayBuffer)
+  dataView.setFloat32(0, value, false)
+  for (let i = 0; i < size; i++) {
+    let t = dataView.getUint8(i).toString(16)
+    res += t == 0 ? '00' : t
+  }
+  return '0x' + res
+}
+
+/**
+ *
+ * @param {*} value
+ * @param {*} size
+ * @returns the negative hex value for a given positive hex value padded with
+ * the given size.
+ */
+function asNegativeHexValue(value, size) {
```

Comment:
This function is being used for floats as well, per above, and for floats this makes absolutely no sense....

I am also not sure why we are doing this transformation on the hex-encoded string for integers.  It seems like for integers (at least 1/2/4-byte ones; 8-byte is complicated) we could just detect that the value is negative and convert `2^{1/2/4} + N`, to hex as a positive value.  For 8-byte integers things are more annoying....  But in practice if the actual value is in the [-2^48, 0] range we can do `2^48 + value`, convert to a 6-byte integer, and pad out with 0xFF at the front.  And if it's outside that range, we should probably throw for now.  Not least because if it picks up another few bits we will stop being able to represent it exactly anyway and all this code will go awry, right?

Reply:
Made changes for this.